File: python/check_if_string_is_a_prefix_of_array_1961.py
from typing import List

class Solution:
    def isPrefixString(self, s: str, words: List[str]) -> bool:
        prefix = ""
        for word in words:
            prefix += word
            if prefix == s:
                return True
            if len(prefix) > len(s):  # Early exit if the prefix is longer than s
                return False
        return False

File: rust/src/check_if_string_is_a_prefix_of_array_1961.rs
use crate::Solution;
impl Solution {
    pub fn is_prefix_string(s: String, words: Vec<String>) -> bool {
        let mut prefix = String::new();

        for word in words {
            prefix.push_str(&word);
            if prefix == s {
                return true;
            }
            if prefix.len() > s.len() {
                return false;
            }
        }

        false
    }
}

